skyportal/handlers/api/mpc.py

```diff
@@ -0,0 +1,197 @@
+                  date:
+                    type: string
+                    description: |
+                      Time to check MPC for.
+                      Defaults to current time.
+                  limiting_magnitude:
+                    type: float
+                    description: |
+                      Limiting magnitude down which to search.
+                      Defaults to 24.0.
+                  search_radius:
+                    type: float
+                    description: |
+                      Search radius for MPC [in arcmin].
+                      Defaults to 1 arcminute.
+
+        responses:
+          200:
+            content:
+              application/json:
+                schema: Success
+          400:
+            content:
+              application/json:
+                schema: Error
+        """
+
+        data = self.get_json()
+        date = data.get('date')
+        if date is None:
+            date = Time.now()
+        else:
+            try:
+                date = Time(arrow.get(date).datetime, format='datetime')
+            except (TypeError, arrow.ParserError):
+                return self.error(f'Cannot parse time input value "{date}".')
+
+        limiting_magnitude = data.get('limiting_magnitude', 24.0)
+        try:
+            limiting_magnitude = float(limiting_magnitude)
+        except Exception:
+            return self.error('Cannot read in limiting magnitude.')
+
+        search_radius = data.get('search_radius', 60.0)
+        try:
+            search_radius = float(search_radius)
+        except Exception:
+            return self.error('Cannot read in search radius.')
+
+        obscode = data.get('obscode', '500')
+
+        with self.Session() as session:
+            obj = session.scalars(
+                Obj.select(session.user_or_token, mode='update').where(Obj.id == obj_id)
+            ).first()
+            if obj is None:
+                return self.error(f"Cannot find object with ID {obj_id}.")
+
+            year = date.strftime("%Y")
+            month = date.strftime("%m")
+            hour = int(date.strftime("%H"))
+            minute = int(date.strftime("%M"))
+            second = float(date.strftime("%S.%f"))
+            day = f"{int(date.strftime('%d'))+(hour+minute/60+second/3600)/24:.2f}"
+
+            ra = Angle(obj.ra, unit='degree')
+            ra_hms = ra.to_string(
+                unit='hourangle', sep=' ', precision=2, pad=True
+            ).split(" ")
+            dec = Angle(obj.dec, unit='degree')
+            dec_dms = dec.to_string(unit=u.deg, sep=' ', precision=1, pad=True).split(
+                " "
+            )
+
+            url = f"{mpcheck_url}?year={year}&month={month}&day={day}&which=pos&ra={ra_hms[0]}+{ra_hms[1]}+{ra_hms[2]}&decl={dec_dms[0]}+{dec_dms[1]}+{dec_dms[2]}&TextArea=&radius={search_radius}&limit={limiting_magnitude}&oc={obscode}&sort=d&mot=h&tmot=s&pdes=u&needed=f&ps=n&type=p"
```

Comment:
Can split this string out to a few short rows.

Reply:
Good point.